Previous patch for 4.2.4 was reworked to fit 4.3.4 and this one is now
available. As the previous comment reports:
this patch fixes a bug into ostream::operator<<(double) due to the wrong size
passed into the __convert_from_v method. The wrong size is then passed to
std::snprintf function, that, on uClibc, doens't handle sized 0 buffer.
It solves bug #1778.

Signed-off-by: Giuseppe Condorelli <giuseppe.condorelli@st.com>
Signed-off-by: Carmelo Amoroso <carmelo.amoroso@st.com>
--------------------------------------------------------------------------------
--- gcc-4.3.4/libstdc++-v3/include/bits/locale_facets.tcc.orig	2009-10-26 16:12:00.000000000 +0100
+++ gcc-4.3.4/libstdc++-v3/include/bits/locale_facets.tcc	2009-10-26 16:14:11.000000000 +0100
@@ -1004,7 +1004,7 @@ _GLIBCXX_BEGIN_LDBL_NAMESPACE
 	const int __cs_size = __fixed ? __max_exp + __prec + 4
 	                              : __max_digits * 2 + __prec;
 	char* __cs = static_cast<char*>(__builtin_alloca(__cs_size));
-	__len = std::__convert_from_v(_S_get_c_locale(), __cs, 0, __fbuf, 
+	__len = std::__convert_from_v(_S_get_c_locale(), __cs, __cs_size, __fbuf,
 				      __prec, __v);
 #endif
 
